driver/remote/driver_windows.go

```diff
@@ -0,0 +1,46 @@
+//go:build windows
+// +build windows
+
+package remote
+
+import (
+	"context"
+	"crypto/tls"
+	"net"
+	"strings"
+
+	"github.com/Microsoft/go-winio"
+	"github.com/pkg/errors"
+)
+
+func (d *Driver) Dial(ctx context.Context) (net.Conn, error) {
+	network, addr, ok := strings.Cut(d.InitConfig.EndpointAddr, "://")
+	if !ok {
+		return nil, errors.Errorf("invalid endpoint address: %s", d.InitConfig.EndpointAddr)
+	}
+
+	dialer := &net.Dialer{}
+
+	var conn net.Conn
+	var err error
+
+	// dial context doesn't support named pipes
+	if network == "npipe" {
+		conn, err = winio.DialPipeContext(ctx, addr)
+	} else {
+		conn, err = dialer.DialContext(ctx, network, addr)
+	}
```

Comment:
This is the only difference?

Can we split this into a helper function, and have that be `dialer_unix.go` and `dialer_windows.go`? We shouldn't duplicate all the rest of this code.

Reply:
Yeah, I expected more divergence between the two but seems that's all that was needed for now. Makes sense to have it as a helper function